device/iot-device-samples/multiplexing-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/MultiplexingSample.java

```diff
@@ -193,22 +122,34 @@ public static void main(String[] args)
         String deviceIdToUnregister = deviceIds.get(0);
 
         System.out.println("Unregistering device " + deviceIdToUnregister + " from multiplexed connection...");
-        multiplexingClient.unregisterDeviceClient(multiplexedDeviceClients.get(deviceIdToUnregister));
+        multiplexClientManager.unregisterDeviceClient(multiplexedDeviceClients.get(deviceIdToUnregister));
         System.out.println("Successfully unregistered device " + deviceIdToUnregister + " from an active multiplexed connection.");
 
         // This code demonstrates how to add a device to an active multiplexed connection without shutting down
         // the whole multiplexed connection or any of the other devices.
         System.out.println("Re-registering device " + deviceIdToUnregister + " to an active multiplexed connection...");
-        multiplexingClient.registerDeviceClient(multiplexedDeviceClients.get(deviceIdToUnregister));
+        multiplexClientManager.registerDeviceClient(multiplexedDeviceClients.get(deviceIdToUnregister));
         System.out.println("Successfully registered " + deviceIdToUnregister + " to an active multiplexed connection");
 
         // Before closing a multiplexing client, you do not need to unregister all of the registered clients.
         // If they are not unregistered, then you can re-open the multiplexing client later and it will still
         // have all of your registered devices
-
         System.out.println("Closing entire multiplexed connection...");
         // This call will close all multiplexed device client instances as well
-        multiplexingClient.close();
+        multiplexClientManager.closeClient();
         System.out.println("Successfully closed the multiplexed connection");
+
+        System.out.println("Shutting down...");
+    }
+
+    private static int acknowledgedSentMessages = 0;
+    private static class TelemetryAcknowledgedEventCallback implements IotHubEventCallback
```

Comment:
Perhaps this belongs on the client wrapper classes we have?

Reply:
I tend to agree that Device can have a wrapper class that implements the TelemetryAck but I am not sure if this logic belongs to the same class. it's especially challenging because we can't implement two separate interfaces in JAVA. Let me do a quick implementation spike and see how it looks like.

Reply:
I think that would complicate the logic in client managers unnecessarily. Let's discuss this further if you feel really strongly about this.